buckets/local/access.go

```diff
@@ -0,0 +1,59 @@
+package local
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/textile/buckets"
+)
+
+// GetPathAccessRoles returns access roles for a path.
+func (b *Bucket) GetPathAccessRoles(ctx context.Context, pth string) (roles map[string]buckets.Role, err error) {
+	ctx, err = b.context(ctx)
+	if err != nil {
+		return
+	}
+	return b.clients.Buckets.GetPathAccessRoles(ctx, b.Key(), pth)
+}
+
+// EditPathAccessRoles updates path access roles by merging the given roles with existing roles
+// and returns the merged roles.
+// roles is a map of string marshaled public keys to path roles. A non-nil error is returned
+// if the map keys are not unmarshalable to public keys.
+// To delete a role for a public key, set its value to buckets.None.
+func (b *Bucket) EditPathAccessRoles(ctx context.Context, pth string, roles map[string]buckets.Role) (merged map[string]buckets.Role, err error) {
+	ctx, err = b.context(ctx)
+	if err != nil {
+		return
+	}
+	err = b.clients.Buckets.EditPathAccessRoles(ctx, b.Key(), pth, roles)
+	if err != nil {
+		return
+	}
+	return b.clients.Buckets.GetPathAccessRoles(ctx, b.Key(), pth)
+}
+
+// PathInvite wraps information needed to collaborate on a bucket path.
+type PathInvite struct {
+	Key  string `json:"key"`
+	Path string `json:"path"`
+}
+
+// SendPathInvite sends a message containing a bucket key and path,
+// which can be used to access a shared file / folder.
+func (b *Bucket) SendPathInvite(ctx context.Context, from thread.Identity, to thread.PubKey, pth string) error {
```

Comment:
For this we were thinking of sending 1 notification per recipient so it would have a list of paths `pths []string`. We also want to customize the body to include some metadata so wondering if we have 1 of the 2 options it would work:
1. Ability to override the body for this message being sent
2. Turn of messaging during the sharing request then we can send the mailbox item directly

Reply:
This is really just a helper function. I decided to not make a single method that edits the access roles and then sends an "invite" to all the new identities because it felt too app specific. For your use case, you can just not use this method and send whatever body message you need, possible with something like this `PathInvite` object. In the end it will be up to your app to handle the invite anyway.